File: native/Footify/app/src/main/java/com/example/footify/model/Player.kt
package com.example.footify.model

import java.util.Date

data class Player(
    val id: String = "",
    val name: String = "",
    val age: Int = 0,
    val position: Position = Position.CENTER_FORWARD,
    val rating: Double = 5.0,
    val shirtNumber: Int = 0,
    val goals: Int = 0,
    val image: String = "",
    val image1: String = "",
    val image2: String = "",
    val createdAt: Date = Date(),
    val updatedAt: Date = Date()
)

enum class Position(val displayName: String) {
    GOALKEEPER("Goalkeeper"),
    RIGHT_BACK("Right Back"),
    LEFT_BACK("Left Back"),
    CENTER_BACK("Center Back"),
    CENTRAL_DEFENSIVE_MIDFIELDER("Central Defensive Midfielder"),
    RIGHT_MIDFIELDER("Right Midfielder"),
    LEFT_MIDFIELDER("Left Midfielder"),
    CENTRAL_MIDFIELDER("Central Midfielder"),
    CENTRAL_ATTACKING_MIDFIELDER("Central Attacking Midfielder"),
    RIGHT_WINGER("Right Winger"),
    LEFT_WINGER("Left Winger"),
    CENTER_FORWARD("Center Forward"),
    STRIKER("Striker");
    
    fun getAbbreviation(): String {
        return when (this) {
            GOALKEEPER -> "GK"
            RIGHT_BACK -> "RB"
            LEFT_BACK -> "LB"
            CENTER_BACK -> "CB"
            CENTRAL_DEFENSIVE_MIDFIELDER -> "CDM"
            RIGHT_MIDFIELDER -> "RM"
            LEFT_MIDFIELDER -> "LM"
            CENTRAL_MIDFIELDER -> "CM"
            CENTRAL_ATTACKING_MIDFIELDER -> "CAM"
            RIGHT_WINGER -> "RW"
            LEFT_WINGER -> "LW"
            CENTER_FORWARD -> "CF"
            STRIKER -> "ST"
        }
    }
}

File: non-native/Footify/src/types/Player.ts
export enum Position {
  GOALKEEPER = 'GOALKEEPER',
  RIGHT_BACK = 'RIGHT_BACK',
  LEFT_BACK = 'LEFT_BACK',
  CENTER_BACK = 'CENTER_BACK',
  CENTRAL_DEFENSIVE_MIDFIELDER = 'CENTRAL_DEFENSIVE_MIDFIELDER',
  RIGHT_MIDFIELDER = 'RIGHT_MIDFIELDER',
  LEFT_MIDFIELDER = 'LEFT_MIDFIELDER',
  CENTRAL_MIDFIELDER = 'CENTRAL_MIDFIELDER',
  CENTRAL_ATTACKING_MIDFIELDER = 'CENTRAL_ATTACKING_MIDFIELDER',
  RIGHT_WINGER = 'RIGHT_WINGER',
  LEFT_WINGER = 'LEFT_WINGER',
  CENTER_FORWARD = 'CENTER_FORWARD',
  STRIKER = 'STRIKER',
}

export interface Player {
  id: string;
  name: string;
  age: number;
  position: Position;
  rating: number;
  shirtNumber: number;
  goals: number;
  image: string;
  image1: string;
  image2: string;
  createdAt: Date;
  updatedAt: Date;
}

export const PositionDisplayNames: Record<Position, string> = {
  [Position.GOALKEEPER]: 'Goalkeeper',
  [Position.RIGHT_BACK]: 'Right Back',
  [Position.LEFT_BACK]: 'Left Back',
  [Position.CENTER_BACK]: 'Center Back',
  [Position.CENTRAL_DEFENSIVE_MIDFIELDER]: 'Central Defensive Midfielder',
  [Position.RIGHT_MIDFIELDER]: 'Right Midfielder',
  [Position.LEFT_MIDFIELDER]: 'Left Midfielder',
  [Position.CENTRAL_MIDFIELDER]: 'Central Midfielder',
  [Position.CENTRAL_ATTACKING_MIDFIELDER]: 'Central Attacking Midfielder',
  [Position.RIGHT_WINGER]: 'Right Winger',
  [Position.LEFT_WINGER]: 'Left Winger',
  [Position.CENTER_FORWARD]: 'Center Forward',
  [Position.STRIKER]: 'Striker',
};

export const PositionAbbreviations: Record<Position, string> = {
  [Position.GOALKEEPER]: 'GK',
  [Position.RIGHT_BACK]: 'RB',
  [Position.LEFT_BACK]: 'LB',
  [Position.CENTER_BACK]: 'CB',
  [Position.CENTRAL_DEFENSIVE_MIDFIELDER]: 'CDM',
  [Position.RIGHT_MIDFIELDER]: 'RM',
  [Position.LEFT_MIDFIELDER]: 'LM',
  [Position.CENTRAL_MIDFIELDER]: 'CM',
  [Position.CENTRAL_ATTACKING_MIDFIELDER]: 'CAM',
  [Position.RIGHT_WINGER]: 'RW',
  [Position.LEFT_WINGER]: 'LW',
  [Position.CENTER_FORWARD]: 'CF',
  [Position.STRIKER]: 'ST',
};

